nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/LayerOpValidation.java

```diff
@@ -1512,19 +1554,24 @@ public void LSTMLayerTestCase1() {
         LSTMLayerOutputs outputs = new LSTMLayerOutputs(sd.rnn.lstmLayer(
                 in, cLast, yLast, null,
                 LSTMLayerWeights.builder()
-                        .weights(sd.var("weights", Nd4j.rand(DataType.FLOAT, nIn, 4 * numUnits)))
-                        .rWeights(sd.var("rWeights", Nd4j.rand(DataType.FLOAT, numUnits, 4 * numUnits)))
-                        .peepholeWeights(sd.var("inputPeepholeWeights", Nd4j.rand(DataType.FLOAT, 3 * numUnits)))
-                        .bias(sd.var("bias", Nd4j.rand(DataType.FLOAT, 4 * numUnits))).build(),
+                        .weights(sd.var("weights", Nd4j.rand(DataType.DOUBLE, nIn, 4 * numUnits)))
+                        .rWeights(sd.var("rWeights", Nd4j.rand(DataType.DOUBLE, numUnits, 4 * numUnits)))
+                        .peepholeWeights(sd.var("inputPeepholeWeights", Nd4j.rand(DataType.DOUBLE, 3 * numUnits)))
+                        .bias(sd.var("bias", Nd4j.rand(DataType.DOUBLE, 4 * numUnits))).build(),
                 c), c);
 
         long[] out = new long[]{bS, numUnits, sL};
         long[] hL = new long[]{bS, numUnits};
         long[] cL = new long[]{bS, numUnits};
 
         assertArrayEquals(out, outputs.getOutput().eval().shape());
-        assertArrayEquals(hL, outputs.getLastTimeStepOutput().eval().shape());
-        assertArrayEquals(cL, outputs.getLastCellStateOutput().eval().shape());
+        assertArrayEquals(hL, outputs.getLastOutput().eval().shape());
+        assertArrayEquals(cL, outputs.getLastState().eval().shape());
+
+        sd.setLossVariables(outputs.getLastOutput(),outputs.getLastTimeStepOutput(),outputs.getTimeSeriesOutput());
+        OpValidation.validate(new TestCase(sd)
+                .gradientCheck(true)
+        );
```

Comment:
As per previous comment: check return value. I won't flag every one with a review comment, so scan through github PR files section to find all the other instances.